core/rawdb/database.go

```diff
@@ -151,12 +151,16 @@ func NewDatabase(db ethdb.KeyValueStore) ethdb.Database {
 // NewDatabaseWithFreezer creates a high level database on top of a given key-
 // value data store with a freezer moving immutable chain segments into cold
 // storage.
-func NewDatabaseWithFreezer(db ethdb.KeyValueStore, freezer string, namespace string, readonly bool) (ethdb.Database, error) {
+func NewDatabaseWithFreezer(db ethdb.KeyValueStore, freezer string, namespace string, args ...bool) (ethdb.Database, error) {
 	// Create the idle freezer instance
-	frdb, err := newFreezer(freezer, namespace, readonly)
+	frdb, err := newFreezer(freezer, namespace, args[0])
```

Comment:
Please try not use some var like `args`, especially the type is bool

Reply:
Do you mean rename this variable?

Reply:
you don't use like args[1] in this function, why do you change the signature of this function

Reply:
wouldn't it be bettter to add another parameter here

Reply:
I'm worried that if adding another parameter obviously, there will be too many places to update beyond this feature, will be kind of messy

Reply:
Done for adding another parameter here to indicate blockprune feature